developing/vcs/master-to-main.md

```diff
@@ -0,0 +1,89 @@
+# [Source Control](./README.md) / Master to Main branch rename
+
+These instructions will help moving repos from using the default branch name `master` and modify it to be
+`main`.
+
+## Instructions to change default branch name
+
+To update the branch name use the npm tool [lgarron/main-branch](https://github.com/lgarron/main-branch):
+
+```sh
+npm install -g main-branch
+main-branch trussworks/master-to-main-test replace master main
+```
+
+Follow up with local modifications:
+
+```sh
+cd git-repo-name
+git pull origin main
+git checkout main
+git branch --delete master
+git branch --set-upstream-to=origin/main main
+git symbolic-ref HEAD refs/heads/main
+git symbolic-ref refs/remotes/origin/HEAD refs/remotes/origin/main
+git fetch --all --prune
+```
+
+Test that the `master` branch has been removed by running this command and getting an error:
+
+```sh
+git checkout master
+```
+
+## Future proofing
+
+Until the `git` tool is modified you'll have to be careful to not creat new repos with `master` as the default
+branch. First make a new git alias:
+
+```sh
+git config --global alias.main '!git init && git checkout -b main'
+git main
+```
+
+You can add an alias to your `bash` or other shell to help:
+
+```sh
+alias git_init='git main'
+```
+
+Alternatively use this alias:
+
+```sh
+alias git_init='git init && git checkout -b main'
+```
+
+Which gives you the ability to run either `git main` or `git_init` to start a new repo with `main` as the default
+branch.
+
+## Github
+
+The branch protection rules in Github will need to be modified for each repo that you touch.
+
+## CircleCI Changes
```

Comment:
Can you look into hosted circleci instances as well? CMS for example is using a private hosted one.

Reply:
I'm thinking that's outside of my scope. Happy to help with it but I'm going to stick to Truss only, non-project changes first. And we can follow up with the CMS folks when we get done with the first part. Same for other projects.

Reply:
I think this is part of the derisking process, but I understand that it should be delayed until after. @jsclarridge maybe you know?